pkg/controller/certificatemanager/certificatemanager.go

```diff
@@ -0,0 +1,286 @@
+package certificatemanager
+
+import (
+	"bytes"
+	"context"
+	"crypto/x509"
+	"errors"
+	"fmt"
+	"time"
+
+	"github.com/openshift/library-go/pkg/crypto"
+	operatorv1 "github.com/tigera/operator/api/v1"
+	"github.com/tigera/operator/pkg/common"
+	"github.com/tigera/operator/pkg/components"
+	"github.com/tigera/operator/pkg/controller/status"
+	"github.com/tigera/operator/pkg/controller/utils/imageset"
+	rmeta "github.com/tigera/operator/pkg/render/common/meta"
+	"github.com/tigera/operator/pkg/tls"
+	"github.com/tigera/operator/pkg/tls/certificatemanagement"
+	corev1 "k8s.io/api/core/v1"
+	kerrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/util/sets"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+)
+
+var (
+	log                    = logf.Log.WithName("tls")
+	ErrInvalidCertDNSNames = errors.New("certificate has the wrong DNS names")
+	errNoPrivateKeyPEM     = errors.New("key pair is missing a private key")
+	errNoCertificatePEM    = errors.New("certificate PEM is missing")
+	errCreatedByOldCA      = errors.New("certificate was created by an older ca")
+)
+
+type certificateManager struct {
+	*x509.Certificate
+	*crypto.CA
+	keyPair *certificatemanagement.KeyPair
+}
+
+// CertificateManager can sign new certificates and has methods to retrieve existing KeyPairs and Certificates. If a user
```

Comment:
I think this might warrant some explanation in terms of just certificates, keys, ca's, and the motivation behind centralising this. I.e. "Tigera components all use tls to communicate and the generated keys / certificates need to be shared among components and generated / managed in K8s <blah blah more explanation>... We also have certificate management which does <blah blah> and requires components to trust the root CA ... all in all having a central place for this drastically reduces the complexity of managing this...".

I hope it goes without saying (even though I'm saying lol) not to use my comment verbatim lol.

Reply:
non-blocking